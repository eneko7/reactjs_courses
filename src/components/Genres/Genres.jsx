import React from 'react';
import propTypes from 'prop-types';
import classnames from 'classnames';
import styles from './Genres.scss';

const Genres = ({ data, classGenres }) => {
  if (data.length > 0) {
    const genres = data.map(item => (
      <span
        className={classnames(styles.genre, styles[classGenres])}
        key={item.toString()}
      >
        {item}
      </span>
    ));
    return (
      <div className={classnames(styles.genres, styles[classGenres])}>
        { genres }
      </div>
    );
  }
  return null;
};

Genres.defaultProps = {
  data: [],
  classGenres: '',
};

Genres.propTypes = {
  data: propTypes.arrayOf(propTypes.string),
  classGenres: propTypes.string,
};

export default Genres;
