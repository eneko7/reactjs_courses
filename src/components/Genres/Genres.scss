@import "../../assets/constants/constants.scss";

.genres {
  display: flex;
  justify-content: flex-start;
  padding: 0 10px 20px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;


  .genre {
    font-family: $fontFamilyRegular;
    margin-right: 10px;
  }
}