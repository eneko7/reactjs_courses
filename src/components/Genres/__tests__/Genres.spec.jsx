import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Genres from '../Genres';

const renderer = new ShallowRenderer();
const data = ['Drama', 'History', 'Comedy'];
const failedData = [];
describe('Genres', () => {
  it('renders correctly', () => {
    renderer.render(<Genres data={data} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('renders with failed data correctly', () => {
    renderer.render(<Genres data={failedData} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
