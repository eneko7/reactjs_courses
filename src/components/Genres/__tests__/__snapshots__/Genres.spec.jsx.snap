// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres renders correctly 1`] = `
<div
  className="genres"
>
  <span
    className="genre"
  >
    Drama
  </span>
  <span
    className="genre"
  >
    History
  </span>
  <span
    className="genre"
  >
    Comedy
  </span>
</div>
`;

exports[`Genres renders with failed data correctly 1`] = `null`;
